New installs from master charm branch builds missing flit_scm wheel and need anyio pinning

https://bugs.launchpad.net/charm-aodh/+bug/2028726

from install hook execution for aodh:

```
...
Processing ./wheelhouse/flit_core-3.9.0.tar.gz
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Processing ./wheelhouse/exceptiongroup-1.1.2.tar.gz
  Installing build dependencies ... error
  error: subprocess-exited-with-error

  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1
  ╰─> [3 lines of output]
      Looking in links: wheelhouse
      ERROR: Could not find a version that satisfies the requirement flit_scm (from versions: none)
      ERROR: No matching distribution found for flit_scm
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```


from install hook execution for placement:

```
2023-07-25 10:01:22 DEBUG unit.placement/0.install logger.go:60 Processing ./wheelhouse/netifaces-0.11.0.tar.gz
2023-07-25 10:01:22 DEBUG unit.placement/0.install logger.go:60   Preparing metadata (setup.py): started
2023-07-25 10:01:24 DEBUG unit.placement/0.install logger.go:60   Preparing metadata (setup.py): finished with status 'done'
2023-07-25 10:01:24 DEBUG unit.placement/0.install logger.go:60 Processing ./wheelhouse/anyio-3.7.1.tar.gz
2023-07-25 10:01:24 DEBUG unit.placement/0.install logger.go:60   Installing build dependencies: started
2023-07-25 10:01:28 DEBUG unit.placement/0.install logger.go:60   Installing build dependencies: finished with status 'error'
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60   error: subprocess-exited-with-error
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60   
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60   × pip subprocess to install build dependencies did not run successfully.
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60   │ exit code: 1
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60   ╰─> [3 lines of output]
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60       Looking in links: wheelhouse
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60       ERROR: Could not find a version that satisfies the requirement setuptools>=64 (from versions: 62.1.0)
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60       ERROR: No matching distribution found for setuptools>=64
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60       [end of output]
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60   
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60   note: This error originates from a subprocess, and is likely not a problem with pip.
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60 error: subprocess-exited-with-error
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60 
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60 × pip subprocess to install build dependencies did not run successfully.
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60 │ exit code: 1
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60 ╰─> See above for output.
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60 
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60 note: This error originates from a subprocess, and is likely not a problem with pip.
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60 Traceback (most recent call last):
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60   File "/var/lib/juju/agents/unit-placement-0/charm/hooks/install", line 8, in <module>
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60     basic.bootstrap_charm_deps()
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60   File "/var/lib/juju/agents/unit-placement-0/charm/lib/charms/layer/basic.py", line 226, in bootstrap_charm_deps
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60     check_call([pip, 'install', '-U', reinstall_flag, '--no-index',
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60   File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60     raise CalledProcessError(retcode, cmd)
2023-07-25 10:01:28 WARNING unit.placement/0.install logger.go:60 subprocess.CalledProcessError: Command '['/var/lib/juju/agents/unit-placement-0/.venv/bin/pip', 'install', '-U', '--force-reinstall', '--no-index', '--no-cache-dir', '-f', 'wheelhouse', 'httpcore==0.17.3', 'certifi==2023.7.22', 'charms.openstack==0.0.1.dev1', 'netaddr==0.7.19', 'tomli==2.0.1', 'Cython==0.29.36', 'dnspython==2.4.0', 'pyaml==21.10.1', 'charmhelpers==1.1.1.dev80', 'charms.reactive==1.5.2', 'sniffio==1.3.0', 'pbr==5.11.1', 'poetry-core==1.6.1', 'flit-core==3.9.0', 'psutil==5.9.5', 'idna==3.4', 'PyYAML==6.0.1', 'netifaces==0.11.0', 'anyio==3.7.1', 'packaging==23.1', 'exceptiongroup==1.1.2', 'h11==0.14.0', 'Jinja2==3.1.2']' returned non-zero exit status 1.
2023-07-25 10:01:29 ERROR juju.worker.uniter.operation runhook.go:1
```

--- a/wheelhouse.txt
+++ b/wheelhouse.txt
@@ -31,6 +31,8 @@ setuptools-scm<=1.17.0;python_version < '3.8'
 # https://github.com/pypa/setuptools_scm/issues/722
 setuptools-scm<7;python_version >= '3.8'
 flit_core<4.0.0;python_version >= '3.8'
+flit_scm<=1.7.0;python_version >= '3.8'
+anyio<3.7.0;python_version >= '3.8'
 charmhelpers>=0.4.0,<2.0.0
 charms.reactive>=0.1.0,<2.0.0
 wheel<0.34;python_version < '3.8'

